### PR TITLE
latest version

### PR DESCRIPTION
**Pull request by nghi-ly**
## What are you changing in this pull request and why?

Beta release for the new **Keep on latest version** setting for the **dbt version** option in an environment's settings.

Renamed file/page title:
- `Upgrade Core version in Cloud` to `Upgrade dbt version in Cloud`

New pages/sections:
- New **Keep on latest version** section in Upgrade Core version in Cloud
- Create new **Version** page
- Release note

Update existing pages: 
- Callout on About dbt Core versions
- Callout on "old" upgrade blog
- Callout on Upgrade to v1.8
- Updates to the Deps page
- Callout on Checking version compatibility page
- Callout on require-dbt-version

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [x] Needs review from product team

Removing pages (delete if not applicable):
- [x] Provide a unique filename for new pages
- [x] Run link testing locally with `npm run build` to update the links that point to deleted pages

Renaming pages:
- [x] Use a unique filename for the new name
- [x] Update YAML front matter in file 
- [x] Update links with new URL
- [x] Add new item to vercel.json (redirects) file
- [x] Update sidebar.js

### CODE REVIEW

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/upgrade-core-in-cloud.md:24]: i just threw this callout in here for grins. do we want to offer a way for people to join the beta?

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/upgrade-core-in-cloud.md]: > You can upgrade to **Keep on latest version** only after you successfully upgrade to 1.7.

is this still accurate?

Comment (jtcohen6) [reply website/docs/docs/dbt-versions/upgrade-core-in-cloud.md]: Good catch - this isn't accurate, there's no prerequisite version you have to be on _before_ trying "Keep on latest version." The only prerequisite is having the feature flag. I'll update the language to clarify.

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (jtcohen6) [inline website/docs/docs/dbt-versions/upgrade-core-in-cloud.md:12]: TODO before merging: update this image with new "pretty" names in dropdown

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (nghi-ly) [reply website/docs/docs/dbt-versions/upgrade-core-in-cloud.md]: thank you!

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (jtcohen6) [reply website/docs/docs/dbt-versions/upgrade-core-in-cloud.md:24]: yes! I updated the text a bit, let me know what you think

**Review by jtcohen6 — COMMENTED** [review]
@nghi-ly Thank you for kicking off this PR!

I have:
- Updated some of the language around `Keep on latest version`. Not set in stone! Definitely interested in hearing your + @azzam34's thoughts
- Started on some changes to related pages: `dbt --version`, `require-dbt-version` / `version-compatibility`
  - Revised the `deps` page that discusses the "package lock" behavior introduced in v1.7. This felt relevant/important, insofar as I'm referencing it from the rationale about no longer checking `require-dbt-version` compatibility. I can spit this set of changes out into a separate PR if it feels necesssary/appropriate.

File: website/blog/2021-11-23-how-to-upgrade-dbt-versions.md
Comment (nghi-ly) [inline website/blog/2021-11-23-how-to-upgrade-dbt-versions.md:26]: sweet message to our dbt-ers!

File: website/docs/docs/dbt-versions/core-upgrade/01-upgrading-to-v1.8.md
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/core-upgrade/01-upgrading-to-v1.8.md:15]: made callout "title" a bit more concise

File: website/docs/docs/dbt-versions/core-versions.md
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/core-versions.md:17]: made text suggs here to keep it "evergreen" (vs upgrade v1.8 guide which is less evergreen).

File: website/docs/reference/commands/deps.md
Comment (nghi-ly) [inline website/docs/reference/commands/deps.md:63]: sugg: shorter title so people can scan faster

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/upgrade-core-in-cloud.md:19]: <3 these updates!

File: website/docs/reference/commands/deps.md
Comment (nghi-ly) [inline website/docs/reference/commands/deps.md:85]: improved example chunking for faster scanning

**Review by nghi-ly — COMMENTED** [review]
thank you so much, @jtcohen6 ! 

i've added my feedback/suggs -- totes not set in stone and open for further edits! 

reviewed everything except for the snippet `_config-dbt-version-check.md`. still mulling ideas over. will work on this part during my early Tues morning.

File: website/docs/docs/dbt-versions/core-versions.md
Comment (jtcohen6) [reply website/docs/docs/dbt-versions/core-versions.md:17]: good by me!

File: website/docs/reference/commands/deps.md
Comment (jtcohen6) [reply website/docs/reference/commands/deps.md:63]: good call!

File: website/docs/reference/commands/deps.md
Comment (jtcohen6) [reply website/docs/reference/commands/deps.md:85]: I like it. For this line in particular, how about just:
```
For other package types, use the `--source` option:
```

File: website/docs/reference/commands/deps.md
Comment (nghi-ly) [inline website/docs/reference/commands/deps.md]:
```suggestion
For other package types, use the `--source` option:
```

File: website/snippets/_config-dbt-version-check.md
Comment (nghi-ly) [inline website/snippets/_config-dbt-version-check.md]: in the spirit of keeping this more evergreen, removed some of the historical context

File: website/docs/reference/global-configs/version-compatibility.md
Comment (nghi-ly) [inline website/docs/reference/global-configs/version-compatibility.md:20]: since this isn't GA-ed yet, went with a callout for the info

i waffled between a callout and subsection -- i think either can work imo

**Review by nghi-ly — COMMENTED** [review]
@jtcohen6 : looks great! thank you! reviewed the snippet `_config-dbt-version-check.md` and the places it's being called

File: website/snippets/_config-dbt-version-check.md
Comment (jtcohen6) [inline website/snippets/_config-dbt-version-check.md]:
```suggestion
Starting in 2024, when you select **Keep on latest version** in dbt Cloud, the `require-dbt-version` config is ignored.
```

File: website/snippets/_config-dbt-version-check.md
Comment (jtcohen6) [inline website/snippets/_config-dbt-version-check.md]: I think we should reorder these bullets, and make the audience clearer for each:
- **For anyone installing packages,** we recommend...
- **For package maintainers,** we recommend writing defensive code. (I should provide an actual example of what I mean here.)

Since I wrote the first draft of this, and I have the context, I can revise this tomorrow and push up changes

File: website/snippets/_config-dbt-version-check.md
Comment (jtcohen6) [reply website/snippets/_config-dbt-version-check.md]: My instinct would be to mention "beta" when we first mention "Keep on latest version" higher up - what do you think?

**Review by jtcohen6 — COMMENTED** [review]
Thanks @nghi-ly, feels like we're getting close!

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (nghi-ly) [reply website/docs/docs/dbt-versions/upgrade-core-in-cloud.md:12]: done! updated screenshot with the new "pretty" names via `pr-deploy` env

File: website/docs/docs/dbt-versions/release-notes/72-Feb-2024/keep-on-latest-version.md
Comment (runleonarun) [inline website/docs/docs/dbt-versions/release-notes/72-Feb-2024/keep-on-latest-version.md:18]: This is such a tiny nit, you can feel free to do a fast follow, but the 1.6 in the screen shot confused me a little. Can we actually show "keep on latest version" in the field and in the pick list?

File: website/docs/reference/global-configs/version-compatibility.md
Comment (runleonarun) [inline website/docs/reference/global-configs/version-compatibility.md]: Should this match other callouts?
```suggestion
:::info Keep on latest version <Lifecycle status='beta' />
```

File: website/docs/reference/global-configs/version-compatibility.md
Comment (runleonarun) [reply website/docs/reference/global-configs/version-compatibility.md:20]: I think it works here and now, but maybe it should be temporary and sometime after GA change it to a section.

File: website/docs/reference/project-configs/require-dbt-version.md
Comment (runleonarun) [inline website/docs/reference/project-configs/require-dbt-version.md:33]: The note below (at line 75) is not closed (no closing `:::`) so it's making the entire section look like a note. 

> :::info Not recommended

File: website/snippets/_config-dbt-version-check.md
Comment (runleonarun) [inline website/snippets/_config-dbt-version-check.md:7]: Should this info also live in the [guide](https://docs.getdbt.com/guides/building-packages?step=1)? and/or package repos eventually? cc @dbeatty10

File: website/snippets/_config-dbt-version-check.md
Comment (runleonarun) [inline website/snippets/_config-dbt-version-check.md:6]: Should this information also live in https://docs.getdbt.com/docs/build/packages#how-do-i-add-a-package-to-my-project?

File: website/snippets/_config-dbt-version-check.md
Comment (runleonarun) [inline website/snippets/_config-dbt-version-check.md:5]: I wonder if instead of having these two bullets of detail here, they should live in the pages I linked to below and we can summarize here and send them to that page for details. Just a thought.

**Review by runleonarun — DISMISSED** [review]
Awesome work here, Ly, capturing so many nuances in so many places. Just a few suggestions. Feel free to move anything to a fast follow.

File: website/docs/reference/commands/version.md
Comment (matthewshaver) [inline website/docs/reference/commands/version.md:2]: We should link to this page [here](https://docs.getdbt.com/reference/global-cli-flags#version) or maybe even remove that section from that page

File: website/docs/reference/commands/version.md
Comment (matthewshaver) [inline website/docs/reference/commands/version.md]:
```suggestion
The `--version` command-line flag returns information about the currently installed version of dbt Core or the dbt Cloud CLI.
```

File: website/docs/reference/commands/version.md
Comment (matthewshaver) [inline website/docs/reference/commands/version.md]:
```suggestion
This flag is not supported when invoking dbt in other dbt Cloud runtimes (for example, the IDE or scheduled runs).
```

File: website/docs/reference/commands/deps.md
Comment (matthewshaver) [inline website/docs/reference/commands/deps.md]: Suggest removing "additional" so it doesn't read like there will be 2 package-lock.yml files in the project
```suggestion
Starting in dbt Core v1.7, dbt generates a `package-lock.yml` file in the root of your project. This contains the complete set of resolved packages based on the `packages` configuration in `dependencies.yml` or `packages.yml`. Each subsequent invocation of `dbt deps` will install from the _locked_ set of packages specified in this file. Storing the complete set of required packages (with pinned versions) in version-controlled code ensures predictable installs in production and consistency across all developers and environments.
```

File: website/docs/reference/commands/deps.md
Comment (matthewshaver) [inline website/docs/reference/commands/deps.md]:
```suggestion
The `package-lock.yml` file includes a `sha1_hash` of the `packages` config. This enables dbt to detect if the `packages` config has been updated, and to rerun dependency resolution. To only check for changes to the `packages` config and update the lock file accordingly without installing those packages, provide the `--lock` flag (that is, `dbt deps --lock`).
```

File: website/docs/reference/commands/deps.md
Comment (matthewshaver) [inline website/docs/reference/commands/deps.md]:
```suggestion
It's possible to force package resolution to rerun, even if the `packages` config hasn't changed, by running `dbt deps --upgrade`. This enables you to get the latest commits from the `main` branch of an internally maintained `git` package while accepting the risk of unpredictable builds. An alternative to running `dbt deps --upgrade` in production is to "ignore" the lock file by adding `package-lock.yml` to your project's `.gitignore` file. If you pursue either approach, dbt Labs strongly recommends adding version pins for third-party packages within your `packages` config.
```

File: website/docs/reference/commands/deps.md
Comment (matthewshaver) [inline website/docs/reference/commands/deps.md]:
```suggestion
The `dbt deps` command can add or update an existing package configuration &mdash; no need to remember the exact syntax for package configurations. 
```

File: website/docs/reference/commands/deps.md
Comment (matthewshaver) [inline website/docs/reference/commands/deps.md]:
```suggestion
For other package types, use the `--source` flag:
```

**Review by jtcohen6 — APPROVED** [review]
Thanks @nghi-ly (& team) for getting this over the finish line! Content looks good to me, I'll let have final say on the last few nits & wrinkles.

File: website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md:12]: It's a little hard to see the "Keep on Latest Version" text. Might be beneficial to grab a screenshot with the text in the main bar rather than the v1.6.

Is there an environment to grab it from that also doesn't have the development view?

File: website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md:19]: Should we add language here about the ability to go from "Keep On Latest Version" back to a version and how ultimately that won't be possible?

File: website/docs/reference/global-configs/version-compatibility.md
Comment (nghi-ly) [reply website/docs/reference/global-configs/version-compatibility.md:20]: exactly! i was thinking the same thing

File: website/docs/reference/project-configs/require-dbt-version.md
Comment (nghi-ly) [reply website/docs/reference/project-configs/require-dbt-version.md:33]: great catch! ty!

File: website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md:16]: Is the "early" part the selling point we want to make? With it, it could read like we're just forcing beta features into their environment, but alternatively, maybe it reads a little more like "you'll always have access to the newest functionality without having to select a version."

```suggestion
By choosing to **Keep on latest version**, you always get the latest fixes and immediate access to new features and functionality for your dbt project. dbt Labs will handle upgrades for you as part of testing and redeploying the dbt Cloud SaaS application.
```

File: website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md]:
```suggestion
This feature is available in beta for select customers, rolling out to wider availability through February and March. If you're interested in early access, please contact your account team to join!
```

File: website/docs/docs/dbt-versions/release-notes/72-Feb-2024/keep-on-latest-version.md
Comment (nghi-ly) [reply website/docs/docs/dbt-versions/release-notes/72-Feb-2024/keep-on-latest-version.md:18]: thanks for flagging! will add this as a fast-follow item

File: website/snippets/_config-dbt-version-check.md
Comment (nghi-ly) [reply website/snippets/_config-dbt-version-check.md:7]: thank you! will add this snippet as a fast-follow

File: website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md
Comment (nghi-ly) [reply website/docs/docs/dbt-versions/upgrade-dbt-version-in-cloud.md:19]: good flag! yes but not for this launch, not yet

File: website/docs/reference/commands/version.md
Comment (nghi-ly) [reply website/docs/reference/commands/version.md:2]: great flag! marking this as a fast-follow

**Review by runleonarun — APPROVED** [review]
🚢 it!

**Pull request merged**
